### PR TITLE
feat: specify disk size and type for dynamic agents in gcp deployment [MLG-224]

### PR DESCRIPTION
**Pull request by drh-determined-ai**
## Description

These changes allow users to specify the [boot disk size and type](https://cloud.google.com/compute/docs/disks) for dynamic agents. [MLG-224]
The parameters are written to a configuration file in the `metadata_startup_script`; this configuration file is used by the provisioner in `determined-master`.

## Test Plan

1. Add an extra volume to the determined-master Docker container command in the metadata_startup_script of the master instance: `-v /path/to/determined-master:/usr/bin/determined-master`. This change should not be committed!
2. Create a deployment with the modification from step 1, including specifications like `--disk-size 159 --disk-type pd-balanced` (the default values are `200` and `pd-standard`). The master health check is _expected to fail_ (because the Docker container can’t find the file specified in the extra volume).
3. Build `determined-master` (i.e., `make -C master build`) with the changes to the provisioner (this must be done on a machine with the same architecture as the deployment master instance, so, e.g., not on a MacBook).
4. Use `gcloud compute scp ...` to move the executable from step 3 to the master instance (the location must match the filepath in step 1).
5. On the master instance, the startup script will have failed. Stop the Docker container and network, and restart the instance.
6. Determined should now be running (you should be able to see the browser UI as usual). Create an experiment. Check for errors in the master log. Check the GCloud Console for dynamic agent instances: they should have the specified disk size and type (see the screenshot in the comments).

## Commentary (optional)

The project id and zone must be included in the disk type URI. Are there potential availability issues, similar to how not all GPUs are available in all zones?

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[MLG-224]: https://hpe-aiatscale.atlassian.net/browse/MLG-224?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by drh-determined-ai** [comment]
Screenshot 2023-03-29 at 1 38 13 PM

File: harness/determined/deploy/gcp/terraform/modules/compute/main.tf [hunk 1]
```diff
@@ -138,6 +140,7 @@ resource "google_compute_instance" "master_instance" {
         --name determined-master-configurator \
         --rm \
         -v /usr/local/determined/etc/:/etc/determined/ \
+        -v /home/danielrhunter/determined-master:/usr/bin/determined-master \
```
Comment (maxrussell) [inline harness/determined/deploy/gcp/terraform/modules/compute/main.tf:143]: Seems like this was accidentally committed.

**Review by maxrussell — CHANGES_REQUESTED** [review]
Still noodling, but leaving this for now.

File: harness/determined/deploy/gcp/terraform/modules/compute/main.tf (hunk 1 above)
Comment (drh-determined-ai) [reply harness/determined/deploy/gcp/terraform/modules/compute/main.tf:143]: Oops!! Thanks for catching this.

**Review by maxrussell — APPROVED** [review]
LGTM

**Pull request merged**